### DISCUSSION TITLE
How to update data get from getServerSideProps ?

### DISCUSSION THREAD

**Discussion by MGMehdi** [vercel/next.js#37194] [Help] [answered]
Hello, 
I'm a new with NextJS and I try to understand how update the data after getServerSideProps ?
For exemple you get a list of user  from the getServerSideProps function and when you update/delete one how can I fetch the new list of user without reloading the page ?

```js
// api
export default async function handler(req, res) {
  const user = await prisma.users.findMany()
  return res.status(200).json(user)
}
```
```js
// index
const Home = ({ users }) => {
  const refreshData = () => {
    // How to refetch new data and save it in 'users' props or else
  }

  return (
    <div>
      <h1>Coucou</h1>
      <button onClick={refreshData}>click</button>
      <ul>
        {users.map((user) => <li key={user.id}>{user.name}</li>)}
      </ul>
    </div>
  )
}

export async function getServerSideProps() {
  const data = await fetch("http://localhost:3000/api/hello")
  const users = await data.json()
  return {
    props: { users }
  }
}

export default Home
```

**Comment by fariasmateuss** [comment]
Hi @MGMehdi 

Try use ISR. See this docs: [Incremental Static Regeneration](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration)

**Reply by MGMehdi** [reply]
Doesn't work like I want because the page need to be reloaded

**Reply by fariasmateuss** [reply]
In that case I recommend using react-query.

**Comment by icyJoseph** [comment, marked as answer]
Take a look at this article! https://www.joshwcomeau.com/nextjs/refreshing-server-side-props/

TL;DR:

```js
import { useRouter } from 'next/router';
function SomePage(props) {
  const router = useRouter();
  // Call this function whenever you want to
  // refresh props!
  const refreshData = () => {
    router.replace(router.asPath);
  }
}
export async function getServerSideProps(context) {
  // Database logic here
}
```

**Reply by Jared-Dahlke** [reply]
seems hacky

**Comment by Antonhansel** [comment]
Hello,

I'm sure I'm missing something, I can't believe that there is no clean way to refresh data originally fetched from the server, `router.replace()` can't be it haha, any pointers ?
